fvm/src/kernel/default.rs

```diff
@@ -110,14 +111,21 @@ where
         let act = state_tree
             .get_actor(addr)?
             .ok_or(anyhow!("state tree doesn't contain actor"))
-            .or_error(ExitCode::ErrIllegalArgument)?;
+            .or_illegal_argument()?;
+
+        if !is_account_actor(&act.code) {
+            return Err(
+                syscall_error!(SysErrIllegalArgument; "target actor is not an account").into(),
```

Comment:
still having a hard time wrapping my head around fvm error handling, but what happens to the error code here? does the calling actor get to decide what to do with the error?

there may be a risk of breaking consensus if that's not the case.